py-polars/polars/internals/dataframe/frame.py

```diff
@@ -161,8 +161,11 @@ class DataFrame:
         * As a list of (name,type) pairs; this is equivalent to the dictionary form.
 
         If you supply a list of column names that does not match the names in the
-        underlying data, the names given here will overwrite them. The number
-        of names given in the schema should match the underlying data dimensions.
+        underlying data, the names given here will overwrite them.
+
+        The number of entries in the schema should match the underlying data
+        dimensions, unless a sequence of dictionaries is being passed, in which case
+        a _partial_ schema can be declared to prevent specific fields from being loaded.
```

Comment:
While I like the partial schema with built-in `select` functionality, this should then also work for other named input data, like dict or a sequence of named Series. Right?

Reply:
Would be clean/consistent; I'll make it happen ;) 

I also spotted a slight (and just resolved) difference between the same init from `pl.from_dicts` - just updated this PR with that fix too. Will do a separate one to bring the other named-input constructors into coherent alignment.